doc/stdenv/stdenv.chapter.md

```diff
@@ -116,6 +116,66 @@ On Linux, `stdenv` also includes the `patchelf` utility.
 
 ## Specifying dependencies {#ssec-stdenv-dependencies}
 
+Build systems often require more dependencies than just what `stdenv` provide. This section describes attributes that `stdenv.mkDerivation` accepts and which can be used to make these dependencies available to the build system.
+
+### Simplified overview
+A full reference of the many kinds of dependencies needed to cover all possible cases is provided in the next section, but here is a simplified overview of the most common kinds. It should cover usual cases.
+
+Add to `nativeBuildInputs` dependencies which are executed during the build:
+- those which are needed on `$PATH` during the build, for example `cmake` and `pkg-config`
+- [setup hooks](#ssec-setup-hooks), for example [`makeWrapper`](#fun-makeWrapper)
+- interpreters needed by [`patchShebangs`](#patch-shebangs.sh) for build scripts (with the `--build` flag), which can be the case for e.g. `perl`
+
+Add to `buildInputs` dependencies which will end up copied or linked into the final output, and used at runtime:
+- libraries used by compilers, for example `zlib`,
+- interpreters needed by [`patchShebangs`](#patch-shebangs.sh) for scripts which are installed, which can be the case for e.g. `perl`
+- runtime dependencies added with [`makeWrapper`](#fun-makeWrapper)
+
+Note that these criteria are not incompatible: if you build a python project that has a build script written in python, python will belong to both `buildInputs` and `nativeBuildInputs`.
```

Comment:
the mention of plugins confuses me more than otherwise.

> maybe python project isn't the best example here since we have an entire infra for just that. maybe?

that's true, do you have suggestions of a replacement that is well-known and yet does not have a big infra in nixpkgs? Maybe perl?

Reply:
Another possiblity is gdk-pixbuf, which is both a lib and a setup hook. but this one is far from obvious though.

Reply:
wayland has to go in both often for wayland-scanner (a program used at build time) and libwayland (linked against; used at runtime).  We _sort of_ hide that by doing `wayland-scanner = wayland.bin` in all-packages.nix.

Reply:
`glib` is a very common example it contains both a library and code generation tools (e.g. `glib-mkenums`).

Reply:
I went for the wayland example, thank you.